lib/syskit/log/datastore/normalize.rb

```diff
@@ -114,14 +116,56 @@ def add_data_block(rt_time, lg_time, raw_data, raw_payload)
                     write raw_data[0, 2]
                     write ZERO_BYTE
                     write raw_data[4..-1]
-                    write raw_payload
+
+                    # Real time are bytes from 0..7
+                    write raw_payload[0..7]
+                    # Logical time are bytes from 8..15
+                    logical_time = extract_logical_time(raw_payload)
+                    if logical_time
+                        write [logical_time.tv_sec].pack("V")
+                        write [logical_time.tv_usec].pack("V")
+                        lg_time = logical_time.microseconds
+                    else
+                        write raw_payload[8..15]
+                    end
+                    write raw_payload[16..-1]
+
```

Comment:
I would guard this whole thing with `if @logical_time_field`, and modify `raw_payload` instead of calling `write` many times.

```
if @logical_time_field
    raw_payload = update_raw_payload_logical_time(raw_payload) 
end
write raw_payload
```